core/domain/stats_jobs_one_off.py

```diff
@@ -18,14 +18,403 @@
+                            of which is stored as a JSON blob.
+                    }
+        """
+        if isinstance(item, stats_models.StartExplorationEventLogEntryModel):
+            yield(
+                item.exploration_id,
+                {'event_type': feconf.EVENT_TYPE_START_EXPLORATION})
+
+        elif isinstance(
+                item, stats_models.CompleteExplorationEventLogEntryModel):
+            yield(
+                item.exploration_id,
+                {
+                    'event_type': feconf.EVENT_TYPE_COMPLETE_EXPLORATION,
+                    'session_id': item.session_id
+                })
+
+        elif isinstance(item, stats_models.StateHitEventLogEntryModel):
+            value = {
+                'event_type': feconf.EVENT_TYPE_STATE_HIT,
+                'version': item.exploration_version,
+                'state_name': item.state_name,
+                'session_id': item.session_id,
+                'created_on': utils.get_time_in_millisecs(item.created_on)
+            }
+            yield(item.exploration_id, value)
+
+        else:
+            value = {
+                'event_type': MigrateStatistics.EVENT_TYPE_STATE_ANSWERS,
+                'version': item.exploration_version,
+                'state_name': item.state_name,
+                'submitted_answer_list': item.submitted_answer_list
+            }
+            yield(item.exploration_id, value)
+
+    @staticmethod
+    def reduce(key, stringified_values):
+        num_starts = 0
+        num_completions = 0
+
+        exp_id = key
+        exploration = exp_services.get_exploration_by_id(exp_id)
+        latest_exp_version = exploration.version
+
+        # Check if model already exists. If it does, delete it.
+        exploration_stats_model = stats_models.ExplorationStatsModel.get_model(
+            exp_id, latest_exp_version)
+        if exploration_stats_model is not None:
+            exploration_stats_model.delete()
+
+        # The list of state hit events in stringified_values.
+        values_state_hit = []
+        # The list of state answers instances in stringified_values.
+        values_state_answer = []
+        # The list of session_ids of learners completing an exploration.
+        completed_session_ids = []
+        for value_str in stringified_values:
+            value = ast.literal_eval(value_str)
+
+            if value['event_type'] == feconf.EVENT_TYPE_START_EXPLORATION:
+                num_starts += 1
+            elif value['event_type'] == feconf.EVENT_TYPE_COMPLETE_EXPLORATION:
+                num_completions += 1
+                completed_session_ids.append(value['session_id'])
+            elif value['event_type'] == feconf.EVENT_TYPE_STATE_HIT:
+                values_state_hit.append(value)
+            elif value['event_type'] == (
+                    MigrateStatistics.EVENT_TYPE_STATE_ANSWERS):
+                values_state_answer.append(value)
+
+        state_stats_mapping = {}
```

Comment:
Add a comment to say that this dict stores cumulative statistics from version 1 to version N of an exploration.